packages/api/get-service-routes.ts

```diff
@@ -0,0 +1,36 @@
+import { APIGatewayEvent, APIGatewayProxyResultV2 } from "aws-lambda";
+import { getServiceStops, ServiceStop, ServiceStops, ServiceStopsQueryInput } from "./client";
+import { ClientError } from "./error";
+import { executeClient } from "./execute-client";
+import { flattenStops } from "./get-service-stops";
+
+export const main = async (event: APIGatewayEvent): Promise<APIGatewayProxyResultV2> =>
+    executeClient(event, getQueryInput, getServiceStops, formatStopsRoutes);
+
+export const getQueryInput = (event: APIGatewayEvent): ServiceStopsQueryInput => {
+    const { pathParameters } = event;
+
+    const serviceId = pathParameters?.serviceId;
+
+    if (!serviceId) {
+        throw new ClientError("Service ID must be provided");
+    }
+
+    if (isNaN(Number(serviceId))) {
+        throw new ClientError("Provided service ID is not valid");
+    }
+
+    return {
+        serviceId: Number(serviceId),
+    };
+};
+
+export const formatStopsRoutes = (stops: ServiceStops): { outbound: ServiceStop[]; inbound: ServiceStop[] } => {
+    const outbound = flattenStops(stops)
+        .filter((stop) => stop.direction === "outbound")
+        .sort((a, b) => Number(a.fromSequenceNumber) - Number(b.fromSequenceNumber));
+    const inbound = flattenStops(stops)
+        .filter((stop) => stop.direction === "inbound")
+        .sort((a, b) => Number(a.fromSequenceNumber) - Number(b.fromSequenceNumber));
+    return { outbound, inbound };
```

Comment:
This is currently returning duplicate stops due to how the data is structured, every link has a `from` and a `to` stop, so the `to` of one link will be the same as the `from` of the next link in the sequence, we could probably get around this by deduping on links that are next to each other in the response with the same `atcoCode`, we can't just dedupe on `atcoCode` globally since certain services might return to the same stop